sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -1783,6 +1784,25 @@ void CAFMaker::produce(art::Event& evt) noexcept {
           FillSBNDCRTTrack(sbndcrttracks[i], srsbndcrttracks.back());
         }
       }
+     
+      // Fill CRT Veto 
+      art::Handle<std::vector<sbnd::crt::CRTVeto>> sbndcrtveto_handle;
+      GetByLabelStrict(evt, fParams.SBNDCRTVetoLabel(), sbndcrtveto_handle);
+      std::vector<art::Ptr<sbnd::crt::CRTVeto>> vetoPtrs;
+      // fill into event
+      if (sbndcrtveto_handle.isValid()) {
+        art::fill_ptr_vector(vetoPtrs, sbndcrtveto_handle);
+	// Only one valid veto per event
+        if (vetoPtrs.size() == 1) {
+  	  // And associated SpacePoint objects
+	  art::FindManyP<sbnd::crt::CRTSpacePoint> spAssoc(sbndcrtveto_handle, evt, fParams.SBNDCRTVetoLabel());
+          if (spAssoc.isValid()) {
+	    // There is one vector of SpacePoints per Veto --> can be empty if no veto condition was satisfied     
+	    const std::vector<art::Ptr<sbnd::crt::CRTSpacePoint>> veto_sp_v(spAssoc.at(vetoPtrs[0].key())); 
```

Comment:
You are asking the `key()` of the _art_ pointer created from the first element of a data product. The value of such key is always `0`.
Also, avoid copying the vector of space point pointers: use a reference.
```suggestion
	    const std::vector<art::Ptr<sbnd::crt::CRTSpacePoint>>& veto_sp_v(spAssoc.at(0)); 
```